plugins/goodreads.koplugin/doublekeyvaluepage.lua

```diff
@@ -0,0 +1,318 @@
+        return
+    end
+    self.page_cnt[1]:setText(curr .. "/" .. total)
+    self.page_cnt.overlap_offset[1] = (self.width - self.page_cnt:getSize().w - 10)
+    self.title_bottom[2] = self.page_cnt
+end
+
+function DoubleKeyValueTitle:onClose()
+    self.kv_page:onClose()
+    return true
+end
+
+local DoubleKeyValueItem = InputContainer:new{
+    key = nil,
+    value = nil,
+    cface_up = Font:getFace("cfont", 22),
+    cface_down = Font:getFace("cfont", 18),
+    width = nil,
+    height = nil,
+    align = "left",
+}
+
+function DoubleKeyValueItem:init()
+    self.dimen = Geom:new{align = "left", w = self.width, h = self.height}
+    if self.callback and Device:isTouchDevice() then
+        self.ges_events.Tap = {
+            GestureRange:new{
+                ges = "tap",
+                range = self.dimen,
+            }
+        }
+    end
+    local key_w = RenderText:sizeUtf8Text(0, self.width, self.cface_down, self.key).x
+    local value_w = RenderText:sizeUtf8Text(0, self.width, self.cface_up, self.value).x
+    if key_w > self.width then
+        self.show_key = RenderText:truncateTextByWidth(self.key, self.cface_down, self.width)
+    else
+        self.show_key = self.key
+    end
+    if value_w > self.width then
+        self.show_value = RenderText:truncateTextByWidth(self.value, self.cface_up, self.width)
+    else
+        self.show_value = self.value
+    end
+    local h = self.dimen.h / 2
+    local w = self.dimen.w
+    self[1] = FrameContainer:new{
+        padding = 10,
+        bordersize = 0,
+        VerticalGroup:new{
+            dimen = Geom:new{ h = h, w = w },
+            padding = 10,
+            LeftContainer:new{
+                padding = 10,
+                dimen = Geom:new{ h = h, w = w },
+                TextWidget:new{
+                    text = self.show_value,
+                    padding = 10,
+                    face = self.cface_up,
+                }
+            },
+            LeftContainer:new{
+                padding = 10,
+                 dimen = Geom:new{ h = h / 5 , w = w },
+                HorizontalSpan:new{ width = Screen:scaleBySize(15), height = 3 }
+            },
+            LeftContainer:new{
+                padding = 10,
+                dimen = Geom:new{ h = h, w = w },
+                TextWidget:new{
+                    text = self.show_key,
+                    padding = 10,
+                    face = self.cface_down,
+                }
+            }
+        }
+    }
+end
+
+function DoubleKeyValueItem:onTap()
+    self.callback()
+    return true
+end
+
+local DoubleKeyValuePage = InputContainer:new{
+    title = "",
+    width = nil,
+    height = nil,
+    show_page = 1,
+    text_input = "",
+    pages = 1,
+    goodreadersKey = "",
+}
+
+function DoubleKeyValuePage:init()
+    self.screen_width = Screen:getSize().w
+    self.screen_height = Screen:getSize().h
+    local settings = G_reader_settings:readSetting("goodreads") or {}
```

Comment:
I suggest shipping goodreaders own setting instead of putting everything into the global reader setting. You can find an example at https://github.com/koreader/koreader/blob/master/frontend/ui/network/manager.lua#L14.